File: leetcode/Question-3.js
// Question - 3 -> Find Duplicate Element on Array 

let array = [1, 2, 3, 4, 5, 2, 5, 4, 1];

function findAllDuplicates(arr) {
    let seen = {};
    let duplicates = [];

    for (let i = 0; i < arr.length; i++) {

        if (seen[arr[i]] === 1) {
            duplicates.push(arr[i]);
        }

        if (seen[arr[i]]) {
            seen[arr[i]]++;
        } else {
            seen[arr[i]] = 1;
        }
    }

    return duplicates;
}

console.log("The Duplicate Element", findAllDuplicates(array));

File: leetcode/Question-3.cpp
// Question - 3 -> Find Duplicate Element on Array

#include <iostream>
using namespace std;

int main()
{
    int arr[] = {1, 4, 6, 4, 1};
    int n = 5;

    cout << "Duplicate elements: ";

    for (int i = 0; i < n; i++)
    {
        for (int j = i + 1; j < n; j++)
        {
            if (arr[i] == arr[j])
            {
                cout << arr[i] << " ";
                break; // so same duplicate prints only once
            }
        }
    }

    return 0;
}
